ts/Series/Treegraph/TreegraphSeries.ts

```diff
@@ -0,0 +1,659 @@
+        state: StatesOptionsKey
+    ): SVGAttributes {
+        if (point && point.isNode) {
+            const { opacity, ...attrs } = Series.prototype.pointAttribs.apply(
+                this,
+                arguments
+            );
+            return attrs;
+        }
+        return super.pointAttribs.apply(this, arguments);
+    }
+
+    /**
+     * Run translation operations for one node.
+     * @private
+     */
+
+    public translateNode(node: TreegraphNode): void {
+        const chart = this.chart,
+            plotSizeY = chart.plotSizeY as number,
+            plotSizeX = chart.plotSizeX as number,
+            { ax, bx, ay, by } = this.layoutModifier,
+            x = ax * node.xPosition + bx,
+            y = ay * node.yPosition + by,
+            symbol = node.symbol,
+            height = node.nodeSizeY,
+            width = node.nodeSizeX,
+            reversed = this.options.reversed,
+            nodeX = chart.inverted ? plotSizeX - width / 2 - x : x - width / 2,
+            nodeY = !reversed ? plotSizeY - y - height / 2 : y - height / 2;
+
+        node.shapeType = 'path';
+        node.nodeX = node.plotX = nodeX;
+        node.nodeY = node.plotY = nodeY;
+        node.shapeArgs = {
+            d: symbols[symbol || 'circle'](nodeX, nodeY, width, height),
+            x: nodeX,
+            y: nodeY,
+            width,
+            height
+        };
+
+        node.shapeArgs.display = node.hasShape() ? '' : 'none';
+        // Calculate data label options for the point
+        node.dlOptions = TreegraphSeries.getDLOptions({
+            level: (this.mapOptionsToLevel as any)[node.level],
+            optionsPoint: node.options
+        });
+        // Pass test in drawPoints
+        node.plotY = 1;
+        // Set the anchor position for tooltips
+        node.tooltipPos = chart.inverted ?
+            [plotSizeY - y, plotSizeX - x] :
+            [x, y];
+    }
+
+    public alignDataLabel(): void {
+        sankeyProto.alignDataLabel.apply(this, arguments);
+    }
+
+    public createNodeColumns(): Array<
+    SankeyColumnComposition.ArrayComposition<TreegraphNode>
+    > {
+        const originalNodes = this.nodes;
+        this.nodes = this.nodes.filter((value: any): boolean => !value.hidden);
+        const nodeColumns = super.createNodeColumns.apply(this, arguments);
+        this.nodes = originalNodes;
+        return nodeColumns as unknown as Array<
+        SankeyColumnComposition.ArrayComposition<TreegraphNode>
+        >;
+    }
+}
+// Handle showing and hiding of the points
+addEvent(TreegraphSeries, 'click', function (e: any): void {
+    const point = e.point as TreegraphNode;
+    point.collapsed = !point.collapsed;
+
+    collapseTreeFromPoint(point, point.collapsed);
+    this.redraw();
+});
+
+function collapseTreeFromPoint(
+    point: TreegraphNode,
+    collapsed: boolean
+): void {
+    point.linksFrom.forEach((link: TreegraphPoint): void => {
+        link.toNode.hidden = collapsed;
```

Comment:
do we need this option with the `toNode.update` in place?

Reply:
This prevents the nodes, that already were collapsed to stay collapsed.